src/streaming/SourceBufferSink.js

```diff
@@ -93,7 +122,12 @@ function SourceBufferSink(mediaSource, mediaInfo, onAppendedCallback) {
     }
 
     function getAllBufferRanges() {
-        return buffer.buffered;
+        try {
+            return buffer.buffered;
+        } catch (e) {
+            log('getAllBufferRanges exception: ' + e.message);
+            return [];
```

Comment:
I'm expecting this function to either return something null-like, or an object implementing the timeranges interface: `.start(n)` `.end(n)` `.length`. Array here works, because the empty array returns length 0 and then nobody calls start or end in that case, but it seems initially confusing reading this to be returning an array. Maybe add an explainer?

Reply:
Null it is then